devices/YarpRobotLoggerDevice/src/YarpTextLoggingUtilities.cpp

```diff
@@ -0,0 +1,226 @@
+    body.local_timestamp = currentTime;
+
+    std::string s;
+    if (message.get(1).isString())
+    {
+        s = message.get(1).asString();
+    } else
+    {
+        return BipedalLocomotion::TextLoggingEntry();
+    }
+
+    yarp::os::Property p(s.c_str());
+
+    if (p.check("level"))
+    {
+        body.text = p.find("message").toString();
+        body.level = p.find("level").toString();
+
+        if (p.check("filename"))
+        {
+            body.filename = p.find("filename").asString();
+        } else
+        {
+            body.filename.clear();
+        }
+
+        if (p.check("line"))
+        {
+            body.line = static_cast<uint32_t>(p.find("line").asInt32());
+        } else
+        {
+            body.line = 0;
+        }
+
+        if (p.check("function"))
+        {
+            body.function = p.find("function").asString();
+        } else
+        {
+            body.function.clear();
+        }
+
+        if (p.check("hostname"))
+        {
+            body.hostname = p.find("hostname").asString();
+        } else
+        {
+            body.hostname.clear();
+        }
+
+        if (p.check("pid"))
+        {
+            body.pid = p.find("pid").asInt32();
+        } else
+        {
+            body.pid = 0;
+        }
+
+        if (p.check("cmd"))
+        {
+            body.cmd = p.find("cmd").asString();
+        } else
+        {
+            body.cmd.clear();
+        }
+
+        if (p.check("args"))
+        {
+            body.args = p.find("args").asString();
+        } else
+        {
+            body.args.clear();
+        }
+
+        if (p.check("thread_id"))
+        {
+            body.thread_id = p.find("thread_id").asInt64();
+        } else
+        {
+            body.thread_id = 0;
+        }
+
+        if (p.check("component"))
+        {
+            body.component = p.find("component").asString();
+        } else
+        {
+            body.component.clear();
+        }
+
+        if (p.check("id"))
+        {
+            body.id = p.find("id").asString();
+        } else
+        {
+            body.id.clear();
+        }
+
+        if (p.check("systemtime"))
+        {
+            body.systemtime = p.find("systemtime").asFloat64();
+        } else
+        {
+            body.systemtime = 0.0;
+        }
+
+        if (p.check("networktime"))
+        {
+            body.networktime = p.find("networktime").asFloat64();
+        } else
+        {
+            body.networktime = body.systemtime;
+            body.yarprun_timestamp.clear();
+        }
+
+        if (p.check("externaltime"))
+        {
+            body.externaltime = p.find("externaltime").asFloat64();
+        } else
+        {
+            body.externaltime = 0.0;
+        }
+
+        if (p.check("backtrace"))
+        {
+            body.backtrace = p.find("backtrace").asString();
+        } else
+        {
+            body.backtrace.clear();
+        }
```

Comment:
Minor. One suggestion would be to have a method to initialize the struct, so that you can get rid of all the ``else`` branches.

Reply:
Done!